pkg/types/v1/structure.go

```diff
@@ -40,16 +41,16 @@ func (st *StructureTest) SetDriverImpl(f func(drivers.DriverConfig) (drivers.Dri
 	st.DriverArgs = args
 }
 
-func (st *StructureTest) RunAll() []*types.TestResult {
+func (st *StructureTest) RunAll(o *output.OutWriter) []*types.TestResult {
```

Comment:
Just a thought, you no longer need to return []*types.TestResult anymore.
All you need for printing the FinalResults is the count of test failed and passed.

I am fine with returning results. You can save some computation by returning the count of pass and fail.

Reply:
good observation, you're right that currently we don't need it, but in the future I plan on supporting output templating and it might be useful to pass around the result objects. I'll leave them for now, but if it turns out I don't need them later on I can just remove it.